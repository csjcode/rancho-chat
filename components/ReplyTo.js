import React from 'react'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'
import colors from '../constants/colors/colors'
import { AntDesign } from '@expo/vector-icons'
import getColors from '../constants/colors/getColors'
const colorsTheme = getColors()

const ReplyTo = (props) => {
  const { text, user, onCancel } = props
  const name = `${user.firstName} ${user.lastName}`

  return (
    <View style={styles.container}>
      <View style={styles.textContainer}>
        <Text numberOfLines={1} style={styles.name}>
          {name}
        </Text>
        <Text numberOfLines={1}>{text}</Text>
      </View>

      <TouchableOpacity onPress={onCancel}>
        <AntDesign name="closecircleo" size={24} color={colorsTheme.blue} />
      </TouchableOpacity>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: colorsTheme.extraLightGrey,
    padding: 8,
    flexDirection: 'row',
    alignItems: 'center',
    borderLeftColor: colorsTheme.blue,
    borderLeftWidth: 4,
  },
  textContainer: {
    flex: 1,
    marginRight: 5,
  },
  name: {
    color: colorsTheme.blue,
    fontFamily: 'medium',
    letterSpacing: 0.3,
  },
})

export default ReplyTo
