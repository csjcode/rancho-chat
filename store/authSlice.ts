import { PayloadAction, createSlice } from "@reduxjs/toolkit";

interface AuthState {
  token: string | null;
  userData: object | null;
  didTryAutoLogin: boolean;
}

const initialState: AuthState = {
  token: null,
  userData: null,
  didTryAutoLogin: false,
};

const authSlice = createSlice({
  name: "auth",
  initialState: {
    token: null,
    userData: null,
    didTryAutoLogin: false,
  } as AuthState,
  reducers: {
    authenticate: (
      state,
      action: PayloadAction<{ token: string; userData: object }>
    ) => {
      const { payload } = action;
      state.token = payload.token;
      state.userData = payload.userData;
      state.didTryAutoLogin = true;
    },
    setDidTryAutoLogin: (state, action) => {
      state.didTryAutoLogin = true;
    },
    logout: (state, action) => {
      state.token = null;
      state.userData = null;
      state.didTryAutoLogin = false;
    },
    updateLoggedInUserData: (
      state,
      action: PayloadAction<{ newData: object }>
    ) => {
      state.userData = { ...state.userData, ...action.payload.newData };
    },
  },
});

export const setDidTryAutoLogin = authSlice.actions.setDidTryAutoLogin;
export const authenticate = authSlice.actions.authenticate;
export const updateLoggedInUserData = authSlice.actions.updateLoggedInUserData;
export const logout = authSlice.actions.logout;
export default authSlice.reducer;
