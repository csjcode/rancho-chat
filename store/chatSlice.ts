import { PayloadAction, createSlice } from "@reduxjs/toolkit";

interface ChatsState {
  chatsData: { [key: string]: any };
}

const initialState: ChatsState = {
  chatsData: {},
};

const chatSlice = createSlice({
  name: "chats",
  initialState,
  reducers: {
    setChatsData: (state, action: PayloadAction<ChatsState>) => {
      state.chatsData = { ...action.payload.chatsData };
    },
  },
});

export const { setChatsData } = chatSlice.actions;
export default chatSlice.reducer;
