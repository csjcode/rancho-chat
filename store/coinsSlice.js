import { createSlice } from '@reduxjs/toolkit'

const coinsSlice = createSlice({
  name: 'coins',
  initialState: {
    storedCoins: {
      coins: ['solana', 'chainlink', 'stepn', 'audius', 'msol'],
    },
  },
  reducers: {
    setStoredCoins: (state, action) => {
      console.log(`action ${JSON.stringify(action)}`)
      const newCoins = action.payload
      state.storedCoins = newCoins
    },
    removeStoredCoin: (state, action) => {
      // {"type":"coins/removeStoredCoin","payload":{"coins":["solana","chainlink","stepn","audius","msol"],"removeCoin":"audius"}}
      console.log(`removeStoredCoin ${JSON.stringify(action)}`)
      const newCoins = action.payload.coins.filter(
        (item) => item !== action.payload.removeCoin,
      )
      // console.log(`action ${JSON.stringify(action)}`)
      // const newCoins = state.storedCoins.coins.pop(action.payload)
      state.storedCoins.coins = newCoins
    },
    addStoredTokenList: (state, action) => {
      console.log(`action ${JSON.stringify(action.payload)}`)
      const newCoins = Array.from(
        new Set([
          ...action.payload.coins,
          ...action.payload.addTokenListStored,
        ]),
      )
      console.log(`newCoins ${newCoins}`)
      state.storedCoins.coins = newCoins
    },
  },
})
export const setStoredCoins = coinsSlice.actions.setStoredCoins
export const removeStoredCoin = coinsSlice.actions.removeStoredCoin
export const addStoredTokenList = coinsSlice.actions.addStoredTokenList
export default coinsSlice.reducer
