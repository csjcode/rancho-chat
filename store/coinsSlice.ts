
import { createSlice } from '@reduxjs/toolkit';

interface StoredCoinsState {
  storedCoins: {
    coins: string[];
  };
}

interface SetStoredCoinsAction {
  payload: StoredCoinsState['storedCoins'];
}

interface RemoveStoredCoinAction {
  payload: {
    coins: string[];
    removeCoin: string;
  };
}

interface AddStoredTokenListAction {
  payload: {
    coins: string[];
    addTokenListStored: string[];
  };
}

const coinsSlice = createSlice({
  name: 'coins',
  initialState: {
    storedCoins: {
      coins: ['solana', 'chainlink', 'stepn', 'audius', 'msol'],
    },
  },
  reducers: {
    setStoredCoins: (state, action: SetStoredCoinsAction) => {
      console.log(`action ${JSON.stringify(action)}`);
      const { payload } = action;
      state.storedCoins = payload;
    },
    removeStoredCoin: (state, action: RemoveStoredCoinAction) => {
      console.log(`removeStoredCoin ${JSON.stringify(action)}`);
      const { payload } = action;
      const { coins, removeCoin } = payload;
      const newCoins = coins.filter((item) => item !== removeCoin);
      state.storedCoins.coins = newCoins;
    },
    addStoredTokenList: (state, action: AddStoredTokenListAction) => {
      console.log(`action ${JSON.stringify(action.payload)}`);
      const { payload } = action;
      const { coins, addTokenListStored } = payload;
      const newCoins = Array.from(
        new Set([...coins, ...addTokenListStored]),
      );
      console.log(`newCoins ${newCoins}`);
      state.storedCoins.coins = newCoins;
    },
  },
});
export const setStoredCoins = coinsSlice.actions.setStoredCoins
export const removeStoredCoin = coinsSlice.actions.removeStoredCoin
export const addStoredTokenList = coinsSlice.actions.addStoredTokenList
export default coinsSlice.reducer
