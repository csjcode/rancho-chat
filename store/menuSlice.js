import { createSlice } from '@reduxjs/toolkit'

const menuSlice = createSlice({
  name: 'menu',
  initialState: {
    storedMenu: {
      map: true,
      tricks: true,
      themeColorLight: false,
    },
  },
  reducers: {
    setStoredMenu: (state, action) => {
      // console.log(`action ${JSON.stringify(action)}`)
      const newMenu = action.payload
      state.storedMenu = newMenu
      // console.log(state.storedMenu)
    },
  },
})
export const setStoredMenu = menuSlice.actions.setStoredMenu
export default menuSlice.reducer
