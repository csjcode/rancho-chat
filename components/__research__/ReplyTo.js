/**
 * 
 * 
 * 
// Unit Tests

import React from 'react'
import { Text, TouchableOpacity, View } from 'react-native'
import { render, fireEvent, waitFor } from '@testing-library/react-native'
import ReplyTo from './ReplyTo'

describe('ReplyTo', () => {
  const mockOnCancel = jest.fn()
  const user = {
    firstName: 'John',
    lastName: 'Doe',
  }

  it('renders correctly', () => {
    const { toJSON } = render(
      <ReplyTo text="Hi" user={user} onCancel={mockOnCancel} />
    )
    expect(toJSON()).toMatchSnapshot()
  })

  it('renders the name and text', () => {
    const { getByText } = render(
      <ReplyTo text="Hi" user={user} onCancel={mockOnCancel} />
    )
    expect(getByText('John Doe')).toBeTruthy()
    expect(getByText('Hi')).toBeTruthy()
  })

  it('calls onCancel when close icon is pressed', async () => {
    const { getByTestId } = render(
      <ReplyTo text="Hi" user={user} onCancel={mockOnCancel} />
    )
    fireEvent.press(getByTestId('close-icon'))
    await waitFor(() => expect(mockOnCancel).toHaveBeenCalled())
  })

  it('does not display the name if user is undefined', () => {
    const { queryByText } = render(<ReplyTo text="Hi" onCancel={mockOnCancel} />)
    expect(queryByText('John Doe')).toBe(null)
  })

  it('does not display the text if text is undefined', () => {
    const { queryByText } = render(
      <ReplyTo user={user} onCancel={mockOnCancel} />
    )
    expect(queryByText('Hi')).toBe(null)
  })

  it('does not display the close icon if onCancel is undefined', () => {
    const { queryByTestId } = render(
      <ReplyTo text="Hi" user={user} />
    )
    expect(queryByTestId('close-icon')).toBe(null)
  })
})

// The tests above ensure that the correct elements are rendered, the correct props are passed in, and that the correct methods are called when appropriate. The first test renders a React Native component and takes a snapshot, which is compared to subsequent snapshots. If a change is made to the rendered component that is not intended, the test will fail. The second test checks that the user's name and text are being rendered correctly. The third test checks that when the close icon is pressed, the mock onCancel function is called. The fourth and fifth tests check that the user's name and text are not rendered if the related props are undefined and that the close icon is not rendered when the onCancel prop is undefined.


 */