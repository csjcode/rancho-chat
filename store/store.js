import { configureStore, getDefaultMiddleware } from '@reduxjs/toolkit'
import { persistReducer, persistStore } from 'redux-persist'

import AsyncStorage from '@react-native-async-storage/async-storage'
import authSlice from './authSlice'
import chatSlice from './chatSlice'
import coinsSlice from './coinsSlice'
import menuSlice from './menuSlice'
import messagesSlice from './messagesSlice'
import notesSlice from './notesSlice'
import userSlice from './userSlice'

const persistNotesConfig = {
  key: 'notes',
  storage: AsyncStorage,
}

const persistedNotes = persistReducer(persistNotesConfig, notesSlice)

export const store = configureStore({
  reducer: {
    auth: authSlice,
    users: userSlice,
    chats: chatSlice,
    messages: messagesSlice,
    menu: menuSlice,
    coins: coinsSlice,
    notes: persistedNotes,
  },
  middleware: getDefaultMiddleware({
    serializableCheck: false,
  }),
})
export const persistor = persistStore(store)
