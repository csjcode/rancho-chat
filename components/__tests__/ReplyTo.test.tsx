import ReactTestRenderer from "react-test-renderer";
import ReplyTo from "../ReplyTo";
import { persistor, store } from "../../store/store";
import { Provider } from "react-redux";

it("renders correctly", async () => {
  const mockProps = {
    text: "This is a test message",
    user: {
      firstName: "John",
      lastName: "Doe",
    },
    onCancel: () => alert("cancel"),
  };
  const tree = ReactTestRenderer.create(
    <Provider store={store}>
      <ReplyTo {...mockProps} />
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
