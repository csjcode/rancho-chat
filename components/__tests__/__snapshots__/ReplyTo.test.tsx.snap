// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#ededed",
      "borderLeftColor": "#3498db",
      "borderLeftWidth": 4,
      "flexDirection": "row",
      "padding": 8,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "marginRight": 5,
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#3498db",
          "fontFamily": "medium",
          "letterSpacing": 0.3,
        }
      }
    >
      John Doe
    </Text>
    <Text
      numberOfLines={1}
    >
      This is a test message
    </Text>
  </View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
</View>
`;
